lib/systems/default.nix

```diff
@@ -421,6 +421,8 @@ let
           isNoStdTarget =
             any (t: hasInfix t final.rust.rustcTarget) ["-none" "nvptx" "switch" "-uefi"];
         };
+      } // {
+        override = allArgs.override or (newArgs: elaborate (allArgs // newArgs));
```

Comment:
nit-picky question:

Would it be better to move `override` up into the same attrs-literal where `canExecute` (for example) is defined?

That'd also avoid needing the `allArgs.override or`, because any `override` present in `args` would be preserved:

```nix
  final =
    {
      override = newArgs: elaborate (allArgs // newArgs);
      # ... parsed, system, config, canExecute, etc...
    }
    // args
    # ...
```

Reply:
> I think we seriously need to consider moving `canExecute` (and all other functions) way down, **after** `// args`

Good point. I guess `override` is one of the few(?) attrs where `args` _should_ win.



> Being explicit is better here, I think.

I figured I'd propose the simpler impl, but I agree with your perspective.

Relying on `//args` could easily be broken by future refactoring; it wouldn't be obvious that `args` preserving its `override` attr was intended; etc.

Therefore I agree. More explicit, more better :grin: